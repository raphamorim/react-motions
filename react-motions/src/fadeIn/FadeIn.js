import React from 'react'
import withFadeIn from './withFadeIn'

function FadeIn({ children }) {
  return (
    <div>
      {withFadeIn(children)}
    </div>
  )
}

export default FadeIn
