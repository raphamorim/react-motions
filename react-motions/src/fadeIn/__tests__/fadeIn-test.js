import React from 'react';
import { mount } from 'enzyme';
import FadeIn from '../fadeIn';

describe('FadeIn', () => {
  it('should render', () => {
    const wrapper = shallow(<FadeIn>FadeIn</FadeIn>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should receive duration prop', () => {
    const wrapper = mount(<FadeIn duration={3}>FadeIn</FadeIn>);

    expect(wrapper.props().duration).toEqual(3);
  });
  it('should receive fillMode prop', () => {
    const wrapper = mount(<FadeIn fillMode="both">FadeIn</FadeIn>);
    expect(wrapper.props().fillMode).toEqual("both");
  });
  it('should render correctly when infinite', () => {
    const wrapper = mount(<FadeIn infinite>FadeIn</FadeIn>);

    expect(wrapper.props().infinite).toBe(true);
  });
});
