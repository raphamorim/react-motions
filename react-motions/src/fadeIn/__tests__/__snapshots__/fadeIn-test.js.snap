// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeIn should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FadeIn
    duration={1}
    fillMode="both"
    infinite={false}
  >
    FadeIn
  </FadeIn>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "FadeIn",
      "style": Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "fadeIn",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "fadeIn",
      },
    },
    "ref": null,
    "rendered": "FadeIn",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "FadeIn",
        "style": Object {
          "WebkitAnimationDuration": "1s",
          "WebkitAnimationFillMode": "both",
          "WebkitAnimationName": "fadeIn",
          "animationDuration": "1s",
          "animationFillMode": "both",
          "animationName": "fadeIn",
        },
      },
      "ref": null,
      "rendered": "FadeIn",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
